### PR TITLE
docs(igxButtonGroup): Add api docs for igxButtonGroup

### PR DESCRIPTION
**Pull request by alexalex972**

### CODE REVIEW

File: projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts [hunk 1]
```diff
@@ -42,34 +42,181 @@ let NEXT_ID = 0;
...
+
+    /**
+     * An @Input property that enables selecting multiple buttons. By default, multiselection is false.
+     * ```html
+     * <igx-buttongroup [multiSelection]="false" [alignment]="alignment"></igx-buttongroup>
+     * ```
+     */
     @Input() public multiSelection = false;
+    /**
+     * An @Input property that allows you the define the values to select.
```
Comment (hanastasov) [inline projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts]: *to define*

File: projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts [hunk 2]
```diff
@@ -42,34 +42,181 @@ let NEXT_ID = 0;
...
+     *```html
+     * <igx-buttongroup #MyChild [multiSelection]="!multi" (onSelect)="onSelect($event)"></igx-buttongroup>
+     *<igx-toast #toast message="You have made a selection!"></igx-toast>
+     *```
+     */
     @Output() public onSelect = new EventEmitter<IButtonGroupEventArgs>();
+
+    /**
+     * An @Ouput property that emits an event when a button is deselected.
+     * An @Ouput property
```
Comment (hanastasov) [inline projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts]: Duplicated line

File: projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts [hunk 3]
```diff
@@ -42,34 +42,180 @@ let NEXT_ID = 0;
...
+     * ```html
+     *<igx-buttongroup [itemContentCssClass]="style1" [multiSelection]="!multi" [values]="alignOptions">
+     *```
+     */
     @Input() set itemContentCssClass(value: string) {
         this._itemContentCssClass = value || this._itemContentCssClass;
     }
+
+    /**
+     * Returns the cssClass of the `igx-buttongroup`.
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts]: @alexalex972 "Returns the CSS class of the item content of the `IgxButtonGroup`"

File: projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts [hunk 4]
```diff
@@ -42,34 +42,180 @@ let NEXT_ID = 0;
...
+     *   let buttonSelect = this.buttonG.itemContentCssClass;
+     *}
+     *```
+     */
     get itemContentCssClass(): string {
         return this._itemContentCssClass;
     }
+
+    /**
+     * An @Input property that enables selecting multiple buttons. By default, multiselection is false.
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts]: @alexalex972 "multiselection" -> "multi-selection"

File: projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts [hunk 5]
```diff
@@ -42,34 +42,180 @@ let NEXT_ID = 0;
...
+     *      })
+     *  ];
+     *  }
+     *  //..
+     * ```
+     * ```html
+     *  <igx-buttongroup [multiSelection]="false" [values]="cities"></igx-buttongroup>
+     * ```
+     */
     @Input() public values: any;
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts]: @alexalex972 We should define a concrete return type. You should contact the person that wrote the code.

File: projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts [hunk 6]
```diff
@@ -42,34 +42,180 @@ let NEXT_ID = 0;
...
+    /**
+     * An @Ouput property that emits an event when a button is selected.
+     *```typescript
+     *@ViewChild("toast")
+     *private toast: IgxToastComponent;
+     *public show(toast){
+     *    this.toast.show();
+     *}
+     *public onSelect(buttongroup){
+     *      this.show(this.toast);
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts]: @alexalex972 `this.toast.show()`

File: projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts [hunk 7]
```diff
@@ -42,34 +42,180 @@ let NEXT_ID = 0;
...
+    /**
+     * An @Ouput property that emits an event when a button is deselected.
+     *```typescript
+     *@ViewChild("toast")
+     *private toast: IgxToastComponent;
+     *public show(toast){
+     *    this.toast.show();
+     *}
+     *public onUnselect(buttongroup){
+     *      this.show(this.toast);
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts]: @alexalex972 `this.toast.show()`

File: projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts [hunk 8]
```diff
@@ -170,7 +171,7 @@ export class IgxButtonGroupComponent implements AfterViewInit {
      *    this.toast.show();
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts]: @alexalex972 Please remove this method.

File: projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts [hunk 9]
```diff
@@ -190,7 +191,7 @@ export class IgxButtonGroupComponent implements AfterViewInit {
...
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts]: @alexalex972 Please remove this method.

File: projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts [hunk 10]
```diff
@@ -42,34 +42,181 @@ let NEXT_ID = 0;
...
+     *  <igx-buttongroup [id]="'igx-dialog-56'" [multiSelection]="!multi" [values]="alignOptions">
+     * ```
+     */
     @HostBinding('attr.id')
     @Input()
     public id = `igx-buttongroup-${NEXT_ID++}`;
 
+    /**
+     * Allows you to set a style using the `itemContentCssClass` input.
+     * The value should be the CSS class name that will be applied to the button group.
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts]: @alexalex972 Can you check whether the class is applied to the button group or it's applied to the item content. If it's to the item content, please update the doc with the correct description.

Comment (alexalex972) [reply projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts]: @mpavlinov I've checked it and it's applied to the igx-button-group.

**Pull request merged**